tests/test_tensor/test_module_spec.py

```diff
@@ -110,28 +122,79 @@ def run_linear_with_spec(label):
     assert tensor_shard_equal(model.weight.grad, model_handy.weight.grad)
     assert tensor_shard_equal(model.bias.grad, model_handy.bias.grad)
 
+def run_check_shared_param():
+    from transformers import BertForMaskedLM, BertConfig
+    hidden_dim = 8
+    num_head = 4
+    sequence_length = 12
+    num_layer = 2
+    vocab_size = 30524
+
+    config = BertConfig(vocab_size=vocab_size,
+                        hidden_size=hidden_dim,
+                        intermediate_size=hidden_dim * 4,
+                        num_attention_heads=num_head,
+                        max_position_embeddings=sequence_length,
+                        num_hidden_layers=num_layer,
+                        hidden_dropout_prob=0.,
+                        attention_probs_dropout_prob=0.)
+    with ColoInitContext(lazy_memory_allocate=False, device=get_current_device()):
+        model = BertForMaskedLM(config)
 
-def run_dist(rank, world_size, port, func):
+    model = model.cuda()
+    parallel_action = ParallelAction(ComputePattern.TP1D)
+    # model.cls.predictions.decoder and model.cls.predictions share the bias, and they should have the same spec
+    assert len(model.cls.predictions.decoder.bias.get_shared_param_modules()) == 2
+    # They are all Linear, so both row is allowed. This should pass check.
+    init_colo_module(model, parallel_action, recursive=True, mode='row')
+    # This should be detected by check because model.cls.predictions.decoder is wrong now.
```

Comment:
I assume you means: 
Setting model.cls.predictions.bias as a TP1DP col spec will leads to the check_colo_module failed. Because its compute pattern is invalid.

Reply:
Because you can not set weight as row while set bias as col?

Reply:
Exactly.